src/main/scala/mesosphere/marathon/MarathonSchedulerService.scala

```diff
@@ -250,140 +188,84 @@ class MarathonSchedulerService @Inject() (
 
   //End Service interface
 
-  //Begin Leader interface, which is required for CandidateImpl.
-  override def onDefeated(): Unit = synchronized {
-    log.info("Defeated (Leader Interface)")
+  //Begin ElectionCandidate interface
 
-    log.info(s"Call onDefeated leadership callbacks on ${leadershipCallbacks.mkString(", ")}")
-    Await.result(Future.sequence(leadershipCallbacks.map(_.onDefeated)), zkTimeout)
-    log.info(s"Finished onDefeated leadership callbacks")
+  def startLeadership(): Unit = synchronized {
+    log.info("As new leader running the driver")
 
-    // Our leadership has been defeated and thus we call the defeatLeadership() method.
-    defeatLeadership()
-  }
-
-  override def onElected(abdicateCmd: ExceptionalCommand[JoinException]): Unit = synchronized {
-    var driverHandlesAbdication = false
-    try {
-      log.info("Elected (Leader Interface)")
+    // execute tasks, only the leader is allowed to
+    migration.migrate()
 
-      //execute tasks, only the leader is allowed to
-      migration.migrate()
+    // start all leadership coordination actors
+    Await.result(leadershipCoordinator.prepareForStart(), config.maxActorStartupTime().milliseconds)
 
-      //run all leadership callbacks
-      log.info(s"""Call onElected leadership callbacks on ${leadershipCallbacks.mkString(", ")}""")
-      Await.result(Future.sequence(leadershipCallbacks.map(_.onElected)), config.onElectedPrepareTimeout().millis)
-      log.info(s"Finished onElected leadership callbacks")
-
-      //start all leadership coordination actors
-      Await.result(leadershipCoordinator.prepareForStart(), config.maxActorStartupTime().milliseconds)
+    // start timers
+    schedulePeriodicOperations()
```

Comment:
Actually, it was called after creating the driver, not running it (the run call is asynchronously). But to avoid regressions, I have moved the `schedulePeriodicOperations` call behind the driver creation again.